### ISSUE TITLE
[Bug]: Emoji's and Tables doesn't work in the Rich Text editor in Appsmith Markdown

### ISSUE THREAD

**Issue by narhzih** [appsmithorg/appsmith#23952]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When using markdown in the Rich text editor in Appsmith, emojis, and tables aren't rendered as expected. 
For example, `:blush:` is supposed to show a blushing emoji, but rather, It just gets rendered as a string. 

The same thing goes for tables; the markdown for the table doesn't render a table. Instead, it renders as an inline string
emojis

### Steps To Reproduce

1. Drag the Rich text editor to the canvas
2. Change to Markdown and insert: 
```
| Month    | Savings |
| -------- | ------- |
| January  | $250    |
| February | $80     |
| March    | $420    |
```

This just gets rendered as an inline string as well. 

### Environment

Production

### Issue video log

https://www.loom.com/share/020632a50a0b4a2ea6b2918307e12a9a

### Version

Cloud & Self - Hosted 1.9.19

**Comment by utkarsh-getsetyo** [comment]
I was also trying out emojis in rich text editor widget. It is not working.

**Comment by Jayavel-S** [comment]
Yes, the same issue here. Adding markdown support in the RTE widget was a good update, but refine it to display the tables properly.

**Comment by Jayavel-S** [comment]
Hi! any updates on this? The tables are not rendering properly.

**Comment by dilippitchika** [comment]
@Jayavel-S no we haven't been able to prioritise this yet. Do you wish to contribute and fix it?

For now you can use a library which converts markdown to html like [showdown](https://showdownjs.com/) and use that as input instead.

**Comment by Jayavel-S** [comment]
I am not an expert with code, so I am not in a position to contribute. I'll check the showdown route you suggested.

**Comment by dilippitchika** [comment]
@Jayavel-S can you let me know why you are trying to show tables in the rich text editor as well? It will help us prioritise better

**Comment by Jayavel-S** [comment]
We use a certain set of internal services that respond in Markdown when called through an API. We would like to display that as a properly rendered and formatted text to the users, who can choose if they need to make any changes before sharing/printing them. 

Since we cannot ask all our users to spend time creating tables and modifying the info from scratch, we need the tables supported. Even if it is possible to do it in just the text widget/any display widget to show the info properly, it would be helpful, as we cannot split/modify the incoming data without introducing much complexity/opting for separate services when we already have an inbuilt one.

Plus, there are a lot of AI-based services too, which are available to provide output in a Markdown format. 

So, this bug is causing us to restrict certain apps from being published to the users.
